### ISSUE TITLE
overloading info object?

### ISSUE THREAD

**Issue by jasmainak** [alphacsc/alphacsc#13]
I saw the following lines of code: https://github.com/alphacsc/alphacsc/blob/master/alphacsc/datasets/somato.py#L130-L137. This seems a little dangerous to me as the `info` object is not meant to store other kinds of data. If you try to do:

```py
from alphacsc.datasets.somato import load_data
from mne.io import write_info

sfreq = 150.
X, info = load_data(sfreq=sfreq, dataset='sample', n_splits=1)
write_info('test.fif', info)
```

and this threw an error. My personal preference is to simply expose the MNE raw objects in the examples but if this helps in making the examples simpler, that's fine too. But perhaps we should find another solution than overloading the `info` object?

**Comment by agramfort** [comment]
I agree with @jasmainak here. People will have an MNE Raw object as input
so we should show them how to work from this

>

**Comment by tomMoral** [comment]
The code is designed to work with `numpy` arrays and `info` just a dictionary. We keep the `mne.Info` structure as it simplifies the plotting functionality but we could store this specific tool in `info['mne_info']` to avoid making it looks like `info` is related to `mne`.

Overall, I don't think this should be changed. But providing tools to format data from `mne` format to our API would definitely be useful. something like `X, info = data_from_mne(raw)`. And maybe renaming `info` to `X_info`.

I use `alphacsc` with other data such as oculometric recording and I'd rather not make it to specific to `mne` format.

**Comment by jasmainak** [comment]
Yes, I am not opposed to using `cdl.transform(X)` with numpy array. But currently, it's a bit opaque for users how to go from an MNE `raw` object to `X`. We do a couple of things in there -- picking channels, applying a window function, and also notch filtering. I think this should be moved into the example rather than inside a function. Otherwise, users won't know how to use CSC on their own data.

By the way, shouldn't there be a way to undo the windowing? Something like: 

```py
z_hat = cdl.transform(X, window)
```

**Comment by agramfort** [comment]
I agree that the example should start from a raw object

**Comment by tomMoral** [comment]
This was closed by #12
